addons/website_forum/models/forum.py

```diff
@@ -142,14 +143,19 @@ def _compute_last_post(self):
     @api.depends('description')
     def _compute_teaser(self):
         for forum in self:
-            if forum.description:
-                desc = forum.description.replace('\n', ' ')
-                if len(forum.description) > 180:
-                    forum.teaser = desc[:180] + '...'
-                else:
-                    forum.teaser = forum.description
-            else:
-                forum.teaser = ""
+            forum.teaser = forum._generate_teaser()
+
+    @api.depends_context('lang')
+    @api.depends('description')
+    def _compute_teaser_non_stored(self):
+        for forum in self:
+            forum.teaser_non_stored = forum._generate_teaser()
+
+    def _generate_teaser(self):
+        if self.description:
+            desc = self.description.replace('\n', ' ')
+            return desc[:180] + '...' if len(desc) > 180 else desc
+        return ""
```

Comment:
This should work too (import `textwrap` on top of course)
```suggestion
        return textwrap.shorten(self.description, 180) if self.description else ""
```